app/assets/yge/redeem_coin.js

```diff
@@ -0,0 +1,69 @@
+window.onload = function() {
+
+    setTimeout(function() {
+        if (!web3.currentProvider || !web3.currentProvider.isMetaMask) {
+            $("step_zero").style.display = "block";
+            $("send_eth").style.display = "none";
+            $("send_eth_done").style.display = "none";
+            $("loading").style.display = "none";
+        } else {
+            $("loading").style.display = "none";
+
+            if (coin_status === 'INITIAL') {
+                $("send_eth").style.display = "block";
+            } else if (coin_status === 'PENDING') {
+                $("send_eth_done").style.display = "block";
+                $("colo_txid").innerHTML = "<a target=new href='https://" + etherscanDomain() + "/tx/" + colo_txid + "'>See your transaction on the blockchain here</a>";
+            }
+        }
+    }, 500);
+};
+
+function redeemCoin() {
+    mixpanel.track("Redeem Coin Click", {});
+    metaMaskWarning();
+
+    var forwarding_address = $("forwarding_address").value.trim();
+
+    // Check for valid checksum address
+    if (!forwarding_address || !(/^(0x)?[0-9a-f]{40}$/.test(forwarding_address) || /^(0x)?[0-9A-F]{40}$/.test(forwarding_address))) {
```

Comment:
Maybe we could use `if (!forwarding_address || !web3.isAddress(forwarding_address)) {` here?